docs/StorageManagement.md

```diff
@@ -0,0 +1,235 @@
+
+The folder will include:
+
+- FybrikStorageAccount types to generate the CRD
+- Open-source implementation of StorageManager APIs
+
+Architecture of StorageManager is based on [Design pattern](https://eli.thegreenplace.net/2019/design-patterns-in-gos-databasesql-package)
+
+It defines `main` that registers various connection types and `plugins` that implement the interface for `AllocateStorage`/`DeleteStorage`. Each plugin registers the connection type it supports in init(). StorageManager invokes the appropriate plugin method based on the registered connection type.
+
+
+## How to support a new connection type
+
+### StorageManager
+
+- Add a new plugin with implementation of `AllocateStorage`/`DeleteStorage` and register it in the main process.
+
+- Create a new docker image of StorageManager
+
+### Fybrik core
+
+- Add a new taxonomy layer describing the connection schema and compile `taxonomy.json`.
+
+- Ensure existence of modules that are able to write/copy to this connection. Update the capabilities in module yamls accordingly.
+
+- Prepare FybrikStorageAccount resources with the shared storage information.
+
+- Update infrastructure attributes related to the storage accounts, e.g., cost.
+
+- Optionally update IT config policies to specify when the new storage can/should be selected
+
+- Optionally extend catalog-connector to support the new connection.
+
+- Deploy new/modified yamls and re-install Fybrik release using StorageManager image and the new taxonomy schema. No change to fybrik_crd is required.
+
+
+## Deployment configuration
+
+In values.yaml add another section `storageManager` with `image` of StorageManager. Modify manager deployment.
+
+
+## Changes to Optimizer and storage type selection
+
+Earlier, the only available storage type was S3. It was hard-coded inside manager as the default connection type used in the write flow of a new dataset. Now, the following changes are required (both to optimizer and the manager naive algorithm):
+
+- add the constraint of storage type/category matching the module protocol
+
+- do not specify the desired connection type and determine it later from the selected storage type
+
+
+## To consider in the future:
+
+- Changes to FybrikApplication (add requirements to the dataset entry)
+
+- Use information about the amount of storage available and amount of data to be written/copied to influence storage selection.
+
+- Extend IT config policies with options for storage management
+
+## Development plan 
+
+### Phase1
+
+- Provide layers for s3, db2, kafka, arrow-flight, what else?
+
+- Changes to FybrikStorageAccount CR
+
+- Implement StorageManager with the defined API for s3 using minio sdk.
+
+- Remove dependency on datashim
+
+- Update documentation accordingly
+
+- Changes to Airbyte module to adapt the suggested taxonomy
+
+
+### Phase2
+
+- Lift the requirement for the default S3 storage, add constraints to the optimizer. Change the non-CSP algorithm as well.
+
```

Comment:
Add additional data store types - db2, kafka, google sheets, postgres?

Reply:
I'd restrict phase2 by a single new data type, and leave others to the phase4.